crates/vm/src/system/memory/paged_vec.rs

```diff
@@ -253,49 +226,109 @@ impl<T: Clone + Default, const PAGE_SIZE: usize> AddressMap<T, PAGE_SIZE> {
             1 << mem_config.pointer_max_bits,
         )
     }
-    pub fn items(&self) -> impl Iterator<Item = (Address, T)> + '_ {
-        self.paged_vecs
-            .iter()
+    pub fn items<F: PrimeField32>(&self) -> impl Iterator<Item = (Address, F)> + '_ {
+        zip_eq(&self.paged_vecs, &self.cell_size)
             .enumerate()
-            .flat_map(move |(as_idx, page)| {
-                page.iter()
-                    .map(move |(ptr_idx, x)| ((as_idx as u32 + self.as_offset, ptr_idx as u32), x))
+            .flat_map(move |(as_idx, (page, &cell_size))| {
+                // TODO: better way to handle address space conversions to F
+                if cell_size == 1 {
+                    page.iter::<u8>()
+                        .map(move |(ptr_idx, x)| {
+                            (
+                                (as_idx as u32 + self.as_offset, ptr_idx as u32),
+                                F::from_canonical_u8(x),
+                            )
+                        })
+                        .collect_vec()
+                } else {
+                    // TEMP
+                    assert_eq!(cell_size, 4);
```

Comment:
Maybe a `CellKind` enum would be better instead of `cell_size`

Reply:
yea I also thought that; I kind of intentionally chose a hacky way to force us to think about it more carefully and change it later